common/dagmcmetadata.cpp

```diff
@@ -0,0 +1,547 @@
+      } else {
+        // failure a volume can only have a single material associated with it
+        std::cout << "more than one material for volume with id " << cellid << std::endl;
+        std::cout << "that does not the the _comp tag associated with it" << std::endl;
+        std::cout << cellid << " has the following material assignments" << std::endl;
+        for ( int j = 0 ; j < material_props.size() ; j++ ) {
+          std::cout << material_props[j] << std::endl;
+        }
+        std::cout << "Please check your material assignments " << cellid << std::endl;
+        exit(EXIT_FAILURE);
+      }
+    }
+    // this is never ok for a volume to have more than one proprety for density
+    if(density_props.size() > 1) {
+      std::cout << "More than one density specified for " << cellid <<std::endl;
+      std::cout << cellid << " has the following density assignments" << std::endl;
+      for ( int j = 0 ; j < density_props.size() ; j++ ) {
+        std::cout << density_props[j] << std::endl;
+      }
+      std::cout << "Please check your density assignments " << cellid << std::endl;
+      exit(EXIT_FAILURE);
+    }
+
+    std::string grp_name = "";
+
+    // determine if we have a density property
+    if (!density_props[0].empty()) {
+      grp_name = "mat:"+material_props[0]+"/rho:"+density_props[0];
+      volume_density_data_eh[eh] = density_props[0];
+    } else {
+      grp_name = "mat:"+material_props[0];
+      volume_density_data_eh[eh] = "";
+    }
+
+    // set the material value
+    volume_material_property_data_eh[eh] = grp_name;
+
+    // not graveyard or vacuum or implicit compliment
+    if (grp_name.find("Graveyard") == std::string::npos && grp_name.find("Vacuum") == std::string::npos
+        && !(DAG->is_implicit_complement(eh)) ) {
+      // set the
+      volume_material_data_eh[eh] = material_props[0];
+    }
+    // found graveyard
+    else if (grp_name.find("Graveyard") != std::string::npos) {
+      volume_material_property_data_eh[eh] = "mat:Graveyard";
+      volume_material_data_eh[eh] = "Graveyard";
+    }
+    // vacuum
+    else if (grp_name.find("Vacuum") != std::string::npos) {
+      volume_material_property_data_eh[eh] = "mat:Vacuum";
+      volume_material_data_eh[eh] = "Vacuum";
+    }
+    // implicit complement
+    else if (DAG->is_implicit_complement(eh)) {
+      if(implicit_complement_material == "" ) {
+        std::cout << "Implicit Complement assumed to be Vacuum" << std::endl;
+        volume_material_property_data_eh[eh] = "mat:Vacuum";
+        volume_material_data_eh[eh] = "Vacuum";
+      } else {
+        volume_material_property_data_eh[eh] = "mat:"+implicit_complement_material;
+        if(implicit_complement_density != "") volume_material_property_data_eh[eh] += "/rho"+implicit_complement_density;
+        volume_material_data_eh[eh] = implicit_complement_material;
+        volume_density_data_eh[eh] = implicit_complement_density;
+      }
+    }
+  }
```

Comment:
There should be something in here telling it what to do if no material is found for a volume